packages/less/test/mocha-playwright/runner.js

```diff
@@ -0,0 +1,203 @@
+            shimMochaInstance(m);
+            delete window.mocha;
+            window.mocha = m
+        },
+        configurable: true
+    })
+
+    Object.defineProperty(window, 'Mocha', {
+        get: function() { return undefined },
+        set: function(m) {
+            shimMochaProcess(m);
+            delete window.Mocha;
+            window.Mocha = m;
+        },
+        configurable: true
+    });
+}
+
+function configureViewport(width, height, page) {
+    if (!width && !height) return page;
+
+    let viewport = page.viewport();
+    width && (viewport.width = width);
+    height && (viewport.height = height);
+
+    return page.setViewport(viewport).then(() => page);
+}
+
+function handleConsole(msg) {
+    const args = msg.args() || [];
+
+    Promise.all(args.map(a => a.jsonValue().catch(() => '')))
+        .then(args => {
+            // process stdout stub
+            let isStdout = args[0] === 'stdout:';
+            isStdout && (args = args.slice(1));
+
+            let msg = util.format(...args);
+            !isStdout && msg && (msg += '\n');
+            process.stdout.write(msg);
+        });
+}
+
+function prepareUrl(filePath) {
+    if (/^[a-zA-Z]+:\/\//.test(filePath)) {
+        // path is URL
+        return filePath;
+    }
+
+    // local path
+    let resolvedPath = path.resolve(filePath);
+    return `file://${resolvedPath}`;
+}
+
+exports.runner = function ({ file, reporter, timeout, width, height, args, executablePath, visible, polling }) {
+    return new Promise(resolve => {
+
+        // validate options
+        if (!file) {
+            throw new Error('Test page path is required.');
+        }
+
+        args = [].concat(args || []).map(arg => '--' + arg);
+        !timeout && (timeout = TIMEOUT_MILLISECONDS);
+        /^\d+$/.test(polling) && (polling = parseInt(polling));
+
+        const url = prepareUrl(file);
+
+        const options = {
+            ignoreHTTPSErrors: true,
+            headless: !visible,
+            executablePath,
+            args
+        };
+
+        const result = chromium.launch(options)
+            .then(browser => browser.newContext()
+                .then(context => context.newPage()
+                    .then(page => {
+                        if (width || height) {
+                            return page.setViewportSize({ width: width || 800, height: height || 600 }).then(() => page);
+                        }
+                        return page;
+                    })
+                    .then(page => {
+                        page.on('console', handleConsole);
+                        page.on('dialog', dialog => dialog.dismiss());
+                        page.on('pageerror', err => console.error(err));
+
+                        return page.addInitScript(initMocha, reporter)
+                            .then(() => page.goto(url))
+                            .then(() => page.waitForFunction(() => window.__mochaResult__, { timeout, polling }))
```

Comment:
Is it possible ```__mochaResult__``` may not be defined after timeout? May want to improve error handling in that scenario so we have a clear message as to why CI failed.

Maybe something like:

```javascript
 .then(result => {
                        if (!result) {
                            throw new Error('Mocha results not found after waiting. The tests may not have run correctly.');
                        }
                        // Close browser before resolving result
                        return browser.close().then(() => result);
                    });
```